wsmaster/codenvy-hosted-api-organization/src/main/java/com/codenvy/organization/api/resource/OrganizationResourcesLimitManager.java

```diff
@@ -0,0 +1,286 @@
+     *
+     * @param organizationId
+     *         organization id
+     * @return resources limits for suborganizations of given parent organization
+     * @throws NullPointerException
+     *         when either {@code organizationId} is null
+     * @throws ServerException
+     *         when any other error occurs
+     */
+    public Page<? extends OrganizationResourcesLimit> getResourcesLimits(String organizationId,
+                                                                         int maxItems,
+                                                                         long skipCount) throws ServerException {
+        requireNonNull(organizationId, "Required non-null organization id");
+
+        return organizationResourcesLimitDao.getByParent(organizationId, maxItems, skipCount);
+    }
+
+    /**
+     * Remove organization resources limit.
+     *
+     * <p>After removing limit given organization won't be able to use parent resources.
+     *
+     * @param organizationId
+     *         organization id
+     * @throws NullPointerException
+     *         when either {@code organizationId} is null
+     * @throws NotFoundException
+     *         when organization with specified id was not found
+     * @throws ConflictException
+     *         when limit can't be changed because suborganization is using existed resources
+     *         or when they are redistributed to next organizations level
+     * @throws ServerException
+     *         when any other error occurs
+     */
+    public void remove(String organizationId) throws ServerException,
+                                                     ConflictException,
+                                                     NotFoundException {
+        requireNonNull(organizationId, "Required non-null organization id");
+
+        try (StripedLocks.WriteLock lock = resourcesLocks.acquiresLock(organizationId)) {
+            checkResourcesAvailability(organizationId, emptyList(), getExistedResources(organizationId));
+            organizationResourcesLimitDao.remove(organizationId);
+        }
+    }
+
+    /**
+     * Checks resources availability of resources in accordance to resources limit changes.
+     *
+     * @param suborganizationId
+     *         identifier of organization which gives new limit
+     * @param newResources
+     *         resources that are defined in new limit
+     * @param existedResources
+     *         resources that are defined in new limit
+     * @throws ConflictException
+     *         when parent organization doesn't have enough resources to increase limit
+     * @throws ConflictException
+     *         when limit can't be changed because suborganization is using existed resources
+     *         or when they are redistributed to next organizations level
+     * @throws ServerException
+     *         when any other error occurs
+     */
+    @VisibleForTesting
+    void checkResourcesAvailability(String suborganizationId,
```

Comment:
What if we change this method to `isAvailable(...)` which doesn't throw any exceptions but returns boolean value instead, along with that we can provide another method similar to existing one, let's say it's `checkAvailable` which will throw the exception? BTW `ForbiddenException` may be good candidate when there is resources context.
The profit that i see, is that if method returns boolean there is a change to write some custom logic around it without 
usage of `try {} catch (ConflictException x) {}`.
Anyway `OrganizationResourcesLimitManager.checkResourcesAvailablity` looks to verbose to me.